libnetwork/drivers/bridge/port_mapping_linux.go

```diff
@@ -66,9 +68,10 @@ func (n *bridgeNetwork) addPortMappings(
 
 	proxyPath := n.userlandProxyPath()
 	disableNAT4, disableNAT6 := n.getNATDisabled()
-	for _, c := range cfg {
-		toBind := make([]portBindingReq, 0, 2)
-		if bindingIPv4, ok := configurePortBindingIPv4(disableNAT4, c, containerIPv4, defHostIP); ok {
+	sortedCfg := sortAndNormPBs(cfg)
+	toBind := make([]portBindingReq, 0, 4)
```

Comment:
I fail to see why this is 4 and not 2 anymore. Either `-p 80` is used, and both `configurePortBinding...` will append in a single iteration, or `-p 0.0.0.0::80 -p '[::]::80'` is used, and each iteration will append only once.

Reply:
I added a comment explaining that, and declared the slice without a size.